File: Curso Logica de Programação Neri Neitzke/Algoritmos em Pascal/pascal/subalgoritmos_funcoes.pas
// Prof Neri Aldoir Neitzke www.INFORMATICON.com.br - videoaulasneri@gmail.com
{Ler 2 números, efetuar as 4 operações matemáticas e mostrar os resultados. c/ parametros e FUNÇÕES}
program subalgoritmos_funcoes;
    var num1, num2, soma, multiplicacao, divisao, subtracao : real;

procedure leitura;
begin
   write('Digite o numero 1 .: ');
   readln(num1);
   write('Digite o numero 2 .: ');
   readln(num2);
end;
function somar(pn1, pn2: real):real;
begin
   soma := pn1 + pn2;
   somar := soma;   //é o return
end;
function multiplicar(pn1, pn2: real):real;
begin
   multiplicacao := pn1 * pn2;
   multiplicar := multiplicacao;   //é o return
end;
function dividir(pn1, pn2: real):real;
begin
   divisao := pn1 / pn2;
   dividir := divisao;   //é o return
end;
function subtrair(pn1, pn2: real):real;
begin
   subtracao := pn1 - pn2;
   subtrair := subtracao;   //é o return
end;
begin
   leitura;
   writeln('A soma e .: ', somar(num1, num2):6:2);
   writeln('A multiplicacao e .: ', multiplicar(num1, num2):6:2);
   writeln('A divisao e .: ', dividir(num1, num2):6:2);
   writeln('A subtracao e .: ', subtrair(num1, num2):6:2);
   write('...................Final do Sistema..................... ');
   readln();
end.

File: Curso Logica de Programação Neri Neitzke/Algoritmos em Java/subalgoritmos_funcoes.java
// Prof Neri Aldoir Neitzke    www.INFORMATICON.com.br - videoaulasneri@gmail.com
//Ler 2 números, efetuar as 4 operações matemáticas e mostrar os resultados.
import javax.swing.JOptionPane;
class subalgoritmos_metodo
{
   static float num1, num2, soma, multiplicacao, subtracao, divisao;
   public static void main(String args[])
   {
      leitura();
      calculos(num1, num2);
      imprimir(multiplicacao, divisao, subtracao,soma);
    }
    public static void leitura()
    {
      num1 = Float.parseFloat(JOptionPane.showInputDialog("Digite o numero 1"));
      num2 = Float.parseFloat(JOptionPane.showInputDialog("Digite o numero 2"));
    }
    public static void calculos(float pn1, float pn2)
    {
      soma = pn1 + pn2;
      multiplicacao = pn1 * pn2;
      divisao = pn1 / pn2;
      subtracao = pn1 - pn2;
    }
    public static void imprimir(float mult, float div, float sub, float som)
    {
       JOptionPane.showMessageDialog(null,"A soma = "+som+ "\nA divisao = "+div+
      "\nA multiplicacao = "+mult+"\nA subtracao = "+sub);
    }
}